controller/src/beerocks/bml/internal/bml_internal.h

```diff
@@ -241,6 +243,11 @@ class bml_internal : public beerocks::socket_thread {
     BML_VAP_INFO *m_vaps                                         = nullptr;
     uint8_t *m_pvaps_list_size                                   = nullptr;
     uint16_t id                                                  = 0;
+
+    // wfa-ca members
+    std::string m_wfa_ca_controller_first_reply;
```

Comment:
As both me and @arnout suggested, please replace the mechanism to use callbacks only which will solve the race conditions and make the code simpler and easier to understand.